doc/querying.md

```diff
@@ -0,0 +1,240 @@
+```
+
+```clojure
+(r/get-select-query (r/distinct users-rel [:id]) {})
+
+=> ["SELECT DISTINCT ON(\"users\".\"id\") \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\""]
+```
+
+### except
+
+Creates a relation that is a combination of two relations with the EXCEPT operator.
+
+EXCEPT example
+
+```clojure
+(-> users-rel
+	(r/except admins-rel)
+	(r/get-select-query {}))
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is-admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM (SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" EXCEPT SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" WHERE \"users\".\"is_admin\" IS TRUE) AS \"users\""]
+```
+
+### extend-with-aggregate
+
+COUNT example:
+
+```clojure
+(require '[com.verybigthings.penkala.helpers :as h])
+
+(-> users-rel
+  (r/extend-with-aggregate :count [:count (h/l 1)])
+  (r/get-select-query {}))
+
+=> ["SELECT count(1) AS \"count\", \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" GROUP BY \"users\".\"id\", \"users\".\"is_admin\", \"users\".\"username\""]
+```
+
+### having
+
+And having operation. If there's already a having clause set, this clause will be joined with AND.
+
+```clojure
+(-> users-rel
+  (r/extend-with-aggregate :count [:count (h/l 1)])
+  (r/having [:> :count (h/l 1)])
+  (r/get-select-query {}))
+
+=> ["SELECT count(1) AS \"count\", \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" GROUP BY \"users\".\"id\", \"users\".\"is_admin\", \"users\".\"username\" HAVING count(1) > 1"]
+```
+
+### intersect
+
+Creates a relation that is a combination of two relations with the INTERSECT operator.
+
+```clojure
+;; ignored by seancorfield/readme
+
+(let [users (:users *env*)]
+  (select! *env* (r/intersect users (r/where users [:is-true :is-admin]))))
+
+=> [#:users{:username test@test.com, :is_admin false, :id 1}]
+```
+
+```clojure
+(-> users-rel
+	(r/intersect admins-rel)
+	(r/get-select-query {}))
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is-admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM (SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" INTERSECT SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" WHERE \"users\".\"is_admin\" IS TRUE) AS \"users\""]
+```
+
+### join
```

Comment:
I see that `:left` is down there? Can we explicitly mention it and list all the options?

Reply:
Yes :+1:

Reply:
Cool. Then just mention it one sentence above the example, or event just make the title a link to the docs.